src/import-export/csv-imp/gnc-tx-import.cpp

```diff
@@ -0,0 +1,960 @@
+    if (orig_lines.size() == 0)
+    {
+        g_set_error (error, GNC_CSV_IMP_ERROR, GNC_CSV_IMP_ERROR_PARSE, "Parsing failed.");
+        return 1;
+    }
+
+    if (guessColTypes)
+    {
+        /* Free column_types if it's already been created. */
+        column_types.clear();
+    }
+    column_types.resize(max_cols, GncTransPropType::NONE);
+
+    if (guessColTypes)
+    {
+        /* Guess column_types based
+         * on the contents of each column. */
+        /* TODO Make it actually guess. */
+    }
+    return 0;
+}
+
+
+/** Convert str into a time64 using the user-specified (import) date format.
+ * @param str The string to be parsed
+ * @param date_format The date format to use.
+ * @return a pointer to a time64 on success, nullptr on failure
+ */
+static time64* convert_date_col_str (const std::string &str, int date_format)
+{
+    auto parsed_date = parse_date (str.c_str(), date_format);
+    if (parsed_date == -1)
+        return nullptr;
+    else
+    {
+        auto mydate = new time64;
+        *mydate = parsed_date;
+        return mydate;
+    }
+}
+
+
+/** Convert str into a gnc_numeric using the user-specified (import) currency format.
+ * @param str The string to be parsed
+ * @param currency_format The currency format to use.
+ * @return a pointer to a gnc_numeric on success, nullptr on failure
+ */
+static gnc_numeric* convert_amount_col_str (const std::string &str, int currency_format)
+{
+    /* If a cell is empty or just spaces return 0 as amount */
+    if(!boost::regex_search(str, boost::regex("[0-9]")))
+        return nullptr;
+
+    auto expr = boost::make_u32regex("[[:Sc:]]");
+    std::string str_no_symbols = boost::u32regex_replace(str, expr, "");
+
+    /* Convert based on user chosen currency format */
+    gnc_numeric val;
+    char *endptr;
+    switch (currency_format)
+    {
+    case 0:
+        /* Currency locale */
+        if (!(xaccParseAmount (str_no_symbols.c_str(), TRUE, &val, &endptr)))
+            return nullptr;
+        break;
+    case 1:
+        /* Currency decimal period */
+        if (!(xaccParseAmountExtended (str_no_symbols.c_str(), TRUE, '-', '.', ',', "\003\003", "$+", &val, &endptr)))
+            return nullptr;
+        break;
+    case 2:
+        /* Currency decimal comma */
+        if (!(xaccParseAmountExtended (str_no_symbols.c_str(), TRUE, '-', ',', '.', "\003\003", "$+", &val, &endptr)))
+            return nullptr;
+        break;
+    }
+
+    auto amount = new gnc_numeric;
+    *amount = val;
+    return amount;
+}
+
+/* Define a class hierarchy to temporarily store transaction/split properties
+ * found in one import line. There is a generic parent class and an implementation
+ * template class. This template class is further specialized for each data type
+ * we support (currently time64, Account, string and gnc_numeric).
+ */
+struct GncTransProperty
+{
+    virtual ~GncTransProperty()
+      //Remove pure designation.
```

Comment:
How about instead of individual parameter objects you create an import struct or std::tuple whose members are a [boost optional](http://www.boost.org/doc/libs/1_61_0/libs/optional/doc/html/index.html) of whatever type is appropriate? As you discover elements in the import you fill in the struct/tuple, then send it off to the transaction-creation routine when done? You could even have a separate struct/tuple for splits and have a vector of those in the transaction tuple to be able to handle an arbitrary number of them.